From 8b297dc5efaf19bde1520f417e323aba0a3fbf92 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kirill=20M=C3=BCller?= <kirill@cynkra.com>
Date: Sun, 8 Dec 2024 06:34:40 +0100
Subject: [PATCH] fix: Fix uninitialized warning

---
 src/duckdb/src/main/connection.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/duckdb/src/main/connection.cpp b/src/duckdb/src/main/connection.cpp
index 4dd7ad11..a08874fc 100644
--- a/src/duckdb/src/main/connection.cpp
+++ b/src/duckdb/src/main/connection.cpp
@@ -28,6 +28,7 @@ Connection::Connection(DatabaseInstance &database)
 }
 
 Connection::Connection(DuckDB &database) : Connection(*database.instance) {
+	warning_cb = nullptr;
 }
 
 Connection::Connection(Connection &&other) noexcept {
-- 
2.43.0

