diff --git a/src/duckdb/src/logging/log_storage.cpp b/src/duckdb/src/logging/log_storage.cpp
index 625757490..253db8514 100644
--- a/src/duckdb/src/logging/log_storage.cpp
+++ b/src/duckdb/src/logging/log_storage.cpp
@@ -37,12 +37,14 @@ StdOutLogStorage::~StdOutLogStorage() {
 
 void StdOutLogStorage::WriteLogEntry(timestamp_t timestamp, LogLevel level, const string &log_type,
                                      const string &log_message, const RegisteredLoggingContext &context) {
+#ifndef DUCKDB_DISABLE_PRINT
 	std::cout << StringUtil::Format(
 	    "[LOG] %s, %s, %s, %s, %s, %s, %s, %s\n", Value::TIMESTAMP(timestamp).ToString(), log_type,
 	    EnumUtil::ToString(level), log_message, EnumUtil::ToString(context.context.scope),
 	    context.context.connection_id.IsValid() ? to_string(context.context.connection_id.GetIndex()) : "NULL",
 	    context.context.transaction_id.IsValid() ? to_string(context.context.transaction_id.GetIndex()) : "NULL",
 	    context.context.thread_id.IsValid() ? to_string(context.context.thread_id.GetIndex()) : "NULL");
+#endif
 }
 
 void StdOutLogStorage::WriteLogEntries(DataChunk &chunk, const RegisteredLoggingContext &context) {
