From a0a4f6f090a3c091344da1c53e919d47ef2051c9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kirill=20M=C3=BCller?= <kirill@cynkra.com>
Date: Mon, 25 Aug 2025 05:57:22 +0200
Subject: [PATCH] disable print for varint

---
 src/duckdb/src/common/varint.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/duckdb/src/common/varint.cpp b/src/duckdb/src/common/varint.cpp
index fdca8149e..ea613af1a 100644
--- a/src/duckdb/src/common/varint.cpp
+++ b/src/duckdb/src/common/varint.cpp
@@ -5,26 +5,32 @@
 namespace duckdb {
 void PrintBits(const char value) {
 	for (int i = 7; i >= 0; --i) {
+#ifndef DUCKDB_DISABLE_PRINT
 		std::cout << ((value >> i) & 1);
+#endif
 	}
 }
 
 void varint_t::Print() const {
 	auto ptr = data.GetData();
 	auto length = data.GetSize();
+#ifndef DUCKDB_DISABLE_PRINT
 	for (idx_t i = 0; i < length; ++i) {
 		PrintBits(ptr[i]);
 		std::cout << "  ";
 	}
 	std::cout << '\n';
+	#endif
 }
 
 void VarintIntermediate::Print() const {
+#ifndef DUCKDB_DISABLE_PRINT
 	for (idx_t i = 0; i < size; ++i) {
 		PrintBits(static_cast<char>(data[i]));
 		std::cout << "  ";
 	}
 	std::cout << '\n';
+#endif
 }
 
 VarintIntermediate::VarintIntermediate(const varint_t &value) {
-- 
2.50.1

